scripts/spyder_win_post_install.py

```diff
@@ -21,17 +21,100 @@
 KEY_C1 = KEY_C0 + r"\command"
 
 
+# ability to run spyder-win-post-install outside of bdist_wininst installer
+# copied from pywin32-win-post-install.py
+# http://pywin32.hg.sourceforge.net/hgweb/pywin32/pywin32/file/default/pywin32_postinstall.py
+try:
+    # When this script is run from inside the bdist_wininst installer,
+    # file_created() and directory_created() are additional builtin
+    # functions which write lines to Python23\pywin32-install.log. This is
+    # a list of actions for the uninstaller, the format is inspired by what
+    # the Wise installer also creates.
+    # https://docs.python.org/2/distutils/builtdist.html#the-postinstallation-script
+    file_created
+    is_bdist_wininst = True
+except NameError:
+    is_bdist_wininst = False # we know what it is not - but not what it is :)
+    # file_created() and directory_created() functions do nothing if post install script
+    # isn't run from bdist_wininst installer, instead if shortcuts and start menu directory
+    # exist, they are removed when the post install script is called with the -remote option
+    def file_created(file):
+        pass
+    def directory_created(directory):
+        pass
+    def get_root_hkey():
+        try:
+            winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE,
+                           root_key_name, 0, winreg.KEY_CREATE_SUB_KEY)

```

Comment:
@ccordoba12 good catch, on windows this key is `hkey_local_machine\Software\Python\PythonCore\2.7` for python installed with msi, however, I don't know what it would be on a windows system that has a bootstrapped Python, eg: WinPython, portable python or any python just unzipped in and executed from a standalone folder.

This is only run by `bdist_wininst` and it's sole purpose is to determine if the user has elevated their rights, if so then use `hkey_local_machine` and if not then use `hkey_current_user`. Therefore even if this call fails, it will just fall back to `hkey_current_user` which is what we want anyway, so it doesn't matter. But for those that have installed official python, and have admin rights, then it will use the system registry keys.